feat: add `no-local-dependencies` rule (#1896)
JoshuaKGoldberg/eslint-plugin-package-json: site/src/content/docs/rules/no-local-dependencies.mdx

---
title: 'no-local-dependencies'

> description: 'Requires that dependencies do not use local file paths, which will cause errors when installing from a registry.'

Author: ```suggestion
description: 'Requires that dependencies do not use local file paths, which will likely result in errors when installing from a registry.'
```

---

{/* end auto-generated rule header */}

import { Tabs, TabItem } from '@astrojs/starlight/components';

This rule enforces that no `dependencies` are declared with local file paths, since this will likely result in errors when installing from a registry.
This includes any paths that start with `file:`, `link:`, `./`, or `../`.

## Examples

<Tabs>
<TabItem label="❌ Incorrect">

```json
{
  "dependencies": {
    "foo": "file:../foo",
    "bar": "link:../bar",
    "baz": "../baz"
  }
}
```

</TabItem>
<TabItem label="✅ Correct">

```json
{
  "dependencies": {
    "foo": "workspace:^",
    "bar": "^1.2.3",
    "baz": "~0.1.2"
  }
}
```

</TabItem>
</Tabs>

## Options

{/* begin auto-generated rule options list */}

| Name            | Description                                                                                 | Type    | Default |
| :-------------- | :------------------------------------------------------------------------------------------ | :------ | :------ |
| `ignorePrivate` | Determines if this rule should be enforced when the package's `private` property is `true`. | Boolean | `true`  |

{/* end auto-generated rule options list */}

### `ignorePrivate`

> When enabled, `ignorePrivate` skips this check entirely.

Author: ```suggestion
When this option is enabled and the package has `"private": true`, the rule skips its check entirely.
```

> This is useful for internal packages that won't be published to npm.

Author: ```suggestion
This can be useful for internal packages that won't be published to a registry.
```

> Disable if you want to enforce for all packages, regardless of privacy.

Author: ```suggestion
Disable it if you want to apply the rule to all packages, regardless of privacy.
```
